R/source.R

```diff
@@ -276,6 +276,19 @@ invoke_cargo <- function(toolchain, specific_target, dir, profile,
 
   tty_has_colors <- isTRUE(cli::num_ansi_colors() > 1L)
 
+  if (identical(.Platform$OS.type, "windows")) {
+    # On Windows, PATH to Rust toolchain should be set by the installer
+    processx_env <- NULL
+  } else {
+    # On Unix-alike, ~/.cargo/bin might not be included in PATH, so we need to
+    # add it here to make sure. It's added to the tail as a fallback, which is
+    # used only when cargo is not found in the user's PATH
+    path_envvar <- Sys.getenv("PATH", unset = "")
+    cargo_path <- path.expand("~/.cargo/bin")
+    # "current" means appending or overwriting the envvars in addition to the current ones.
+    processx_env <- c("current", PATH = glue("{path_envvar}:{cargo_path}"))
+  }
+
```

Comment:
What if we insert here a check for `cargo` using `Sys.which()`? If I understand correctly, if `Sys.which("cargo")` works, then `processx::run()` also works fine. If that is so, perhaps something like `if (all(nchar(Sys.which("cargo")) == 0)) ui_w("Can't find {.code cargo} on the PATH...")`?

Reply:
> if `Sys.which("cargo")` works, then `processx::run()` also works fine.

Unfortunately, no. On CRAN machine, I saw `cargo` is in `PATH` but `rustc` is not (I really don't understand why...).

Reply:
Anyway, I guess your point is that if we tweak `PATH`, we should let the user know so that the user can notice the possible problem on their PATH setting, right?

Reply:
Ok, so does your PR fix this issue on CRAN as well? Or only for regular users?

Reply:
> Anyway, I guess your point is that if we tweak PATH, we should let the user know so that the user can notice the possible problem on their PATH setting, right?

My point was that if we **fix** the path and **still** cannot find `cargo` or `rustc`, then perhaps it is useful to notify the user, like "hey we did our best but still there is no cargo, your build may fail".

Reply:
> does your PR fix this issue on CRAN as well?

Yes. gifski package uses this way: https://github.com/r-rust/gifski/blob/6b86cc6b60abbc2294db821f27cae37413df70c2/src/Makevars#L12-L13

My package sources `~/.cargo/env`, which basically sets `PATH`.

>  notify the user, like "hey we did our best but still there is no cargo, your build may fail".

I see.

Reply:
I think your warnings do not respect `is_queit`. When it is set to `TRUE`, we should get **no** notifications except for errors thrown with `ui_throw()`.

Reply:
Thanks for catching!